src/main/java/org/apache/maven/plugins/checkstyle/Violation.java

```diff
@@ -0,0 +1,153 @@
+package org.apache.maven.plugins.checkstyle;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+
+/**
+ * Holds data about a single violation and represents the violation itself.
+ */
+public class Violation
+{
+  private final String source;
+
+  private final String file;
+
+  private final int line;
+
+  private @Nullable Integer column;
+
+  private final String severity;
+
+  private final String message;
+
+  private final String ruleName;
+
+  private final String category;
+
+  // Leaving out column, because there is no CHECKSTYLE:OFF support.
```

Comment:
This feels like the tail wagging the dog. If there's a good reason to put column here, checkstyle is what should go.

Reply:
Please change accordingly. My commit is not about checkstyle rule changes for this project  There is a reason why 7 parameters are the maximum, and I, as a new committer, do not feel in the position to chabge that rule or to add //CHECKSTYLE: OFF statements.

If you are unhappy with the existing rules, please do not ask first time committers to change them.  Maybe you even want the rule to be more mutable? Maybe fiddle with the messages before printing?
I'd create new rules for this, but your coding style seems different than mine.

That said, change as you like.